File: mobile-client/android/UseCase/Profile/Detail/LoadProfileDetailUseCase.kt
package app.mobiling.client.usecase.profile.detail

import app.mobiling.client.contract.profile.detail.ProfileDetailPayload
import app.mobiling.client.data.profile.detail.ProfileDetailGateway

class LoadProfileDetailUseCase(
    private val gateway: ProfileDetailGateway,
) {
    suspend operator fun invoke(profileId: String): ProfileDetailPayload = gateway.loadProfileDetail(profileId)
}

File: mobile-client/ios/UseCase/Profile/Detail/LoadProfileDetailUseCase.swift
public struct LoadProfileDetailUseCase {
    private let gateway: ProfileDetailGateway

    public init(gateway: ProfileDetailGateway) {
        self.gateway = gateway
    }

    public func callAsFunction(profileId: String) async throws -> ProfileDetailPayload {
        try await gateway.loadProfileDetail(profileId: profileId)
    }
}
